sdk/eventhub/event-hubs/src/diagnostics/tracing.ts

```diff
@@ -52,3 +53,41 @@ export function createMessageSpan(
     kind: SpanKind.PRODUCER
   });
 }
+
+/**
+ * Converts TryAddOptions into the modern shape (OperationOptions) when needed.
+ * (this is something we can eliminate at the next major release of EH _or_ when
+ * we release with the GA version of opentelemetry).
+ *
+ * @internal
+ */
+export function convertTryAddOptionsForCompatibility(tryAddOptions: TryAddOptions): TryAddOptions {
+  /* eslint-disable-next-line @typescript-eslint/ban-ts-comment */
+  // @ts-ignore: parentSpan is deprecated and this is compat code to translate it until we can get rid of it.
+  const possibleParentSpan = tryAddOptions.parentSpan;
+
+  if (!possibleParentSpan) {
+    // assume that the options are already in the modern shape.
+    return tryAddOptions;
+  }
+
+  const convertedOptions: TryAddOptions = {
+    ...tryAddOptions,
+    tracingOptions: {
+      spanOptions: {
+        parent: isSpan(possibleParentSpan) ? possibleParentSpan.context() : possibleParentSpan
+      }
+    }
```

Comment:
If in the future, we add more things to `OperationTracingOptions` than just spanOptions and the user has set those, then the above will override that and make the convertedOptions have tracingOptions with just spanOptions.

Also, if both the parentSpan and tracingOptions are set by the user, the above will respect the parentSpan. Shouldnt we go the other way around?

Reply:
> If in the future, we add more things to `OperationTracingOptions` than just spanOptions and the user has set those, then the above will override that and make the convertedOptions have tracingOptions with just spanOptions.

Yeah, I see what you're saying. Keep in mind that this particular path is when the user is using the 'v1' of this interface (ie, the way we shipped GA with) which has literally the single parentSpan field on it. So there'd be no expansion possible once we're going down this route. 

> 
> Also, if both the parentSpan and tracingOptions are set by the user, the above will respect the parentSpan. Shouldnt we go the other way around?

The user would have to be doing something deliberate interesting. They'd both be filling out parentSpan (and ignoring the deprecated warning) _and_ they'd have to construct the needed items for tracingOptions, which is swiftly going to be incompatible with that approach anyways. 

The only reason I did the copying on line 75 was just to say "well, we could legitimately add more items to AddSpanOptions that are actually useful and not tracing related" so I didn't want to strip them out.

Reply:
> we could legitimately add more items to AddSpanOptions

You mean TryAddOptions?

> Yeah, I see what you're saying. Keep in mind that this particular path is when the user is using the 'v1' of this interface (ie, the way we shipped GA with) which has literally the single parentSpan field on it. So there'd be no expansion possible once we're going down this route.

It is possible if one does the "deliberately interesting thing" of setting both the parentSpan and the tracingOptions in TryAddOptions.

Can we go with this in the initial check?
```js
if (!possibleParentSpan || tryAddOptions.tracingOptions) {
    // assume that the options are already in the modern shape.
    return tryAddOptions;
  }
```

Reply:
Done.